plugins/completion/cmp/sources/cmp-tabby.nix

```diff
@@ -11,37 +11,59 @@ in
 {
   meta.maintainers = [ maintainers.GaetanLepage ];
 
-  options.plugins.cmp-tabby = helpers.neovim-plugin.extraOptionsOptions // {
-    host = helpers.defaultNullOpts.mkStr "http://localhost:5000" ''
-      The address of the tabby host server.
-    '';
+  # TODO: introduced 24-06-18, remove after 24.11
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "cmp-tabby"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (mkRenamedOptionModule (basePluginPath ++ [ "extraOptions" ]) settingsPath)
+      (mkRenamedOptionModule (basePluginPath ++ [ "host" ]) (settingsPath ++ [ "host" ]))
+      (mkRenamedOptionModule (basePluginPath ++ [ "maxLines" ]) (settingsPath ++ [ "max_lines" ]))
+      (mkRenamedOptionModule (basePluginPath ++ [ "runOnEveryKeyStroke" ]) (
+        settingsPath ++ [ "run_on_every_keystroke" ]
+      ))
+      (mkRenamedOptionModule (basePluginPath ++ [ "stop" ]) (settingsPath ++ [ "stop" ]))
```

Comment:
This could be mapped through `helpers.toSnakeCase` like mkNeovimPlugin does. This is fine though as it is small, simple and readable.

Off topic: Isn't this a good use-case for mk*Plugin accepting a path instead of a namespace?